packages/manager/modules/vps/src/dashboard/tile/configuration/upgrade/routing.js

```diff
@@ -0,0 +1,222 @@
+import get from 'lodash/get';
+import set from 'lodash/set';
+
+import { Environment } from '@ovh-ux/manager-config';
+import VpsConfigurationTile from '../service';
+import { ORDER_TRACKING_URLS } from './constants';
+
+import component from './component';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('vps.detail.dashboard.configuration.upgrade', {
+    url: '/upgrade/{upgradeType:memory|storage}?upgradeStatus&upgradeOrderId',
+    layout: 'ouiModal',
+    redirectTo: (transition) => {
+      const $q = transition.injector().get('$q');
+      const upgradeTypePromise = transition.injector().getAsync('upgradeType');
+      const configurationTilePromise = transition
+        .injector()
+        .getAsync('configurationTile');
+
+      return $q
+        .all({
+          upgradeType: upgradeTypePromise,
+          configurationTile: configurationTilePromise,
+        })
+        .then(({ upgradeType, configurationTile }) => {
+          const upgradeToPlan = get(
+            configurationTile,
+            `upgrades[${upgradeType}].plan`,
+          );
+
+          // check if the upgrade is available
+          if (!upgradeToPlan) {
+            return 'vps.detail.dashboard';
+          }
+
+          const upgradeModelPlan = get(
+            configurationTile,
+            `model[${upgradeType}]`,
+          );
+
+          // check if the model is setted correctly => if we access to the state
+          // directly (without clicking on oui-radio in dashboard state)
+          if (upgradeModelPlan.planCode !== upgradeToPlan.planCode) {
+            set(configurationTile.model, upgradeType, upgradeToPlan);
```

Comment:
To me it is not a good idea to update the value of `configurationTile.model` especially the value being injected. This is causing side effects and updating the injection without the programmer/user knowing. 
What is the need ?

Reply:
I would like to reset the model in order to have the previous value in case of modal closing (click on cancel button, click on dismiss button or click outside of the modal). It was possible to reload the state but as we have a lot of call to make, it was not very fluid for customer. So I use this approach to directly reset the models that control the toggle group present in parent state view (the dashboard view).

Reply:
Could it be possible to create a copy when entering the modal, update that copy and if you need to on modal close set that copy as the new model for its parent ?

Reply:
Mmmmmh, I don't really understand why changing the reference of the model object will change. The parent model object reference will also change.